src/backend/core/models.py

```diff
@@ -1123,6 +1123,33 @@ def __str__(self):
         return f"{self.thread} - {self.mailbox} - {self.role}"
 
 
+class ThreadEvent(BaseModel):
+    """Thread event model to store different types of events in a thread."""
+
+    thread = models.ForeignKey(
+        "Thread", on_delete=models.CASCADE, related_name="events"
+    )
+    type = models.CharField(_("type"), max_length=36)
+    channel = models.ForeignKey(
+        "Channel",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="thread_events",
+        help_text=_("Channel that created this event"),
+    )
+    data = models.JSONField(_("data"), default=dict, blank=True)
+
+    class Meta:
+        db_table = "messages_threadevent"
+        verbose_name = _("thread event")
+        verbose_name_plural = _("thread events")
+        ordering = ["created_at"]
+
+    def __str__(self):
+        return f"{self.thread} - {self.type}"
+
+
```

Comment:
In case I would like to create an event which relies on a specific message ? (e.g: Ask a review), the `message_id` will be stored into the `data` payload ?
I feel this model will have lot of entities so if we need to lookup on some fields those ones should be indexed and easy to access ?